### ISSUE TITLE
Feature Request: Being able to 'pin' a proximity chat so users can move their avatars as they talk

### ISSUE THREAD

**Issue by alexandria-p** [thecodingmachine/workadventure#1304]
A big issue my colleagues and I found as we were using Work Adventure is that we'd organically bump into each other in proximity chat (max. 4 at a time), idle and chat for a bit then decide to move together as a group to find someone else or to go to a conference room.

Only issue is, we lose our chat as our avatars move as a bit of a rag-tag group.

Perhaps an option to 'pin' the members who are currently engaged in a proximity chat so that the members involved can move around on-screen and still continue talking? 

- Concept of toggling to and from 'proximity' chat
- Create a toggle onscreen to Toggle proximity chat to 'closed'/'private'/'pinned' [so it's no longer a proximity chat]
- Turning the toggle to 'on', all current members in proximity will move to closed chat.
- Users can individually leave the closed chat. It is disbanded when there is only one person remaining & that person will be toggled back to proximity chat again.
- Members in a closed chat will not be able to join new proximity chats until they un-pin the one they are in first by toggling back to 'proximity' - so if you want to chat with new people in your proximity, you will have to leave the closed chat.
- As usual, conference rooms etc will take precedence over the closed chat (and will immediately return all members to the 'proximity' default)
- Will require a clear visual indication that users have toggled from proximity to 'closed' or 'private', for players involved and to other players not involved
- If a player in a closed chat stands in proximity with one or more players in an open proximity chat for a given length of time, there should be a prompt to the player to ask if they want to toggle back to proximity chat.

Questions this raises: 
- Can this feature be abused by people who don't want to be involved in the closed chat? 
- Should there be a voting system to move a proximity chat to a closed chat? 
- Should the chat toggle immediately to closed , but all users are notified and are given a clear prompt for opting-out/toggling back to proximity ?
- Should there be a 'group leader' of the closed chat (whoever first proposed it), and if they choose to disband the chat all members are toggled back to proximity?

Any further ideas welcomed 👍

**Comment by Lurkars** [comment]
Another approach for the problem is #791. In general your idea sounds cool, but complicated. I know that WA devs always want to keep things as simple as possible (KISS). So I personally also like the "Auto-Follow" approach more, because it also solves another problem: The following of "I show you something on the map". Combined with the new "Block User" function, auto-follow stalkers should then not be a problem anymore and it also is a more real world approach, since you can keep talking to each other by following and not by "pinning" a conversation, also other can still join the conversation, which is also like in RL.

**Comment by jonnytest1** [comment]
> Another approach for the problem is #791. In general your idea sounds cool, but complicated. I know that WA devs always want to keep things as simple as possible (KISS). So I personally also like the "Auto-Follow" approach more, because it also solves another problem: The following of "I show you something on the map". Combined with the new "Block User" function, auto-follow stalkers should then not be a problem anymore and it also is a more real world approach, since you can keep talking to each other by following and not by "pinning" a conversation, also other can still join the conversation, which is also like in RL.

i dont think auto-follow is easier to implement than pinning a conversation
(its not even the same as pet following since pets dont have collisions)

**Comment by Lurkars** [comment]
> i dont think auto-follow is easier to implement than pinning a conversation
> (its not even the same as pet following since pets dont have collisions)
>
> and since you probably want to keep the conversation going when switching maps - irl its gonna be a pin anyways

I don't meant with "complicated" regarding implementation, more about UX. There is a bunch logical requirements and settings mentioned for this feature. Of course it is a different feature, that may solve other use cases, as Auto-follow also do. But the main problem of loosing conversation while moving could also be solved with just Auto-Follow to pin conversation, no logic for keeping others out, add a different conv. type etc. So in UX there would just be one option: Auto-follow on/off and I think it is nearer to RL experience then like implementing "long distance calls".

Also I like Auto-follow for other reasons. Switching maps would result in both switching maps on Auto-follow, so don't see a problem here.

**Comment by alexandria-p** [comment]
I am glad to hear that there is a feature in the works that will help with moving groups around! Glad that I raised a feature request before going ahead and forking the project without an understanding of what is in the feature pipeline.

There may still be value in pinning a proximity chat for free movement but I am hyper-aware that overall long-distance chat  shouldn't replace proximity chat, as it removes the point of Work Adventure which is to connect virtually as you would in an office (popping your head into someone's cubicle, etc).

If pinning a chat were to be implemented, it is a bit of a complex endeavour and there are several logical problems that would need to be considered & addressed, so I do agree that the auto-follow functionality will solve the use-case I have raised (groups moving together & wanting to remain in proximity chat) 😄

**Comment by Lurkars** [comment]
@alexandria-p I am not sure what is in the pipeline. I am only community developer and not related to The Coding Machine and just wrote down my thoughts on this. The Auto-follow is also just an idea of the community. So it is still up to the real devs to decide if any of this get's implemented or merged if provided by community.

**Comment by Kharhamel** [comment]
Auto follow isn't too bad to implement in practice, the game already listens to the position changes of every player next to you, so all you have to do calculate the movement vector and apply to your own character. It is a bit wonky because of the latency between your target's move and yours but it still works. 
Alternatively I could try accelerating the player toward its target but it might be harder to notify the server of your changes of position.

To me, it is more of a UI issue. How do you activate it and how do you desactivate it?
Following another player takes away soem of the control you have on your own character, so it needs to be activated by something very explicit (I think about a menu attached to a user on right click with an option 'Follow')

And for breaking the follow, it needs to be easy and quick for the same reason (people don't like loosing control and wants to get it back easily). I though about breaking the follow when you input a movement , but this means that you cannot use your movement keys to adjust your position/trajectory while still following. And clicking on a contextual menu might be not ergonomic because the followed player will be moving.

**Comment by moufmouf** [comment]
:+1: for the autofollow option (already said that).

I agree it is more of a UI issue than anything else.

> Following another player takes away soem of the control you have on your own character, so it needs to be activated by something very explicit (I think about a menu attached to a user on right click with an option 'Follow')

Now that I think about it, I wonder if we should not, on the contrary, be triggered by the person who wants to "lead" the party.

I mean, quite often, if someone is new to WorkAdventure, and you want him/her to follow you, it will be hard to explain: "ok, right click on me and click the "follow" button".

Instead, we could have a big "Lead" button (I'm not sure about the wording) that would appear when in a bubble. Anyone inside the bubble would automatically auto-follow your character.
As long as it is easy to stop following someone (by touching any input key), I think we should be ok. Most users are well intended, and we already have a ban/ignore feature that can be used to ignore people that would be abusing the feature.

Thoughts?

**Comment by Kharhamel** [comment]
That means that anyone can displace someone absent/not focused on wa. That sounds like a troll's heaven

**Comment by Lurkars** [comment]
yes, I also think that follow should not be controlled by others, if someone wants to lead the party, there is already a prox. chat running and saying: "Follow me! (Please click on the follow action)" is enough control for party leader. If this action is too hard, then other features like blocking another user or triggering emotes is also too hard, which I think isn't.

**Comment by alexandria-p** [comment]
Would you follow them by clicking their avatar and selecting 'follow' from somewhere, or would you be able to see a 'follow' button in proximity chat on their video feed, like with the Report/Block flag?

**Comment by Kharhamel** [comment]
I don't know what would be more convenient. The best solution would be to implement both and then monitor the usages.

Now I am thinking about something else. What would happen if 2 users try to follow each other? We will need to prevent this scenario

**Comment by Lurkars** [comment]
As Visit-card is already feature Clicking on user, I stay with my suggestion using the round menu with multiple options like "show visit card", "follow" etc. When following will be just cancelled with using movement controls, then there is no problem with 2 following each other, because as soon as one moves, he leaves following the otherone which automatically follows.

**Comment by ValdoTR** [comment]
Can we close that now that we have the "follow me" feature?
